app/endpoints/nwb_validation.py

```diff
@@ -0,0 +1,144 @@
+        detail={
+            "code": ApiErrorCode.BAD_REQUEST,
+            "detail": f"NWB validation failed: {error_output}",
+        },
+    )
+
+
+async def _process_nwb(file: UploadFile, temp_file_path: str) -> None:
+    """Validate nwb file with pynwb using asyncio subprocess."""
+    command_args = ["pynwb-validate", temp_file_path]
+
+    try:
+        process = await asyncio.create_subprocess_exec(
+            command_args[0],
+            *command_args[1:],
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+
+        stdout_data, stderr_data = await process.communicate()
+
+        stdout = stdout_data.decode().strip()
+        stderr = stderr_data.decode().strip()
+
+        # Check the return code
+        if process.returncode != 0:
+            _raise_validation_error(file, stderr, stdout, process.returncode)
+
+    except FileNotFoundError as e:
+        L.error(f"Validation tool not found: {e!s}")
+        # Catch exceptions if 'pynwb-validate' is not in the PATH
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail={
+                "code": ApiErrorCode.BAD_REQUEST,
+                "detail": "Required validation tool 'pynwb-validate' not found.",
+            },
+        ) from e
+    except Exception as e:
+        L.error(f"Unexpected Nwb error validating file {file.filename}: {e!s}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail={
+                "code": ApiErrorCode.BAD_REQUEST,
+                "detail": f"An unexpected error occurred during validation: {e!s}",
+            },
+        ) from e
+    else:
+        return
+
+
+async def _validate_and_read_nwb_file(file: UploadFile) -> tuple[bytes, str]:
+    """Validates file extension and reads content."""
+    L.info(f"Received file upload: {file.filename}")
+    allowed_extensions = {".nwb"}
+    file_extension = f".{file.filename.split('.')[-1].lower()}" if file.filename else ""
+
+    if not file.filename or file_extension not in allowed_extensions:
+        L.error(f"Invalid file extension: {file_extension}")
+        valid_extensions = ", ".join(allowed_extensions)
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail={
+                "code": ApiErrorCode.BAD_REQUEST,
+                "detail": f"Invalid file extension. Must be one of {valid_extensions}",
+            },
+        )
+
+    content = await file.read()
+    if not content:
+        _handle_empty_file(file)
+
+    return content, file_extension
+
+
+def activate_test_nwb_endpoint(router: APIRouter) -> None:
+    """Define neuron file test endpoint."""
+
+    @router.post(
+        "/test-nwb-file",
+        summary="Validate new format.",
+        description="Tests a new file (.nwb) with basic validation.",
+    )
+    async def test_nwb_file(
```

Comment:
It seems that the last commits introduced quite a lot of changes compared to the previous code, and this endpoint has been reverted to be async, but it should be sync as discussed previously.
Also the name was changed to be not "test" but "validate".